darts/ad/anomaly_model/forecasting_am.py

```diff
@@ -0,0 +1,673 @@
+from darts.logging import get_logger, raise_if_not
+from darts.models.forecasting.forecasting_model import ForecastingModel
+from darts.timeseries import TimeSeries
+
+logger = get_logger(__name__)
+
+
+class ForecastingAnomalyModel(AnomalyModel):
+    def __init__(
+        self,
+        model: ForecastingModel,
+        scorer: Union[AnomalyScorer, Sequence[AnomalyScorer]],
+    ):
+        """Forecasting-based Anomaly Detection Model
+
+        The forecasting model may or may not be already fitted. The underlying assumption is that `model`
+        should be able to accurately forecast the series in the absence of anomalies. For this reason,
+        it is recommend to either provide a model that has already been fitted and evaluated to work
+        appropriately on a series without anomalies, or to ensure that a simple call to the :func:`fit()`
+        method of the model will be sufficient to train it to satisfactory performance on a series without anomalies.
+
+        Calling :func:`fit()` on the anomaly model will fit the underlying forecasting model only
+        if ``allow_model_training`` is set to ``True`` upon calling ``fit()``.
+        In addition, calling :func:`fit()` will also fit the fittable scorers, if any.
+
+        Parameters
+        ----------
+        model
+            An instance of a Darts forecasting model.
+        scorer
+            One or multiple scorer(s) that will be used to compare the actual and predicted time series in order
+            to obtain an anomaly score ``TimeSeries``.
+            If a list of `N` scorer is given, the anomaly model will call each
+            one of the scorers and output a list of `N` anomaly scores ``TimeSeries``.
+        """
+
+        raise_if_not(
+            isinstance(model, ForecastingModel),
+            f"Model must be a darts ForecastingModel not a {type(model)}.",
+        )
+        self.model = model
+
+        super().__init__(model=model, scorer=scorer)
+
+    def fit(
+        self,
+        series: Union[TimeSeries, Sequence[TimeSeries]],
+        past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        allow_model_training: bool = False,
+        forecast_horizon: int = 1,
+        start: Union[pd.Timestamp, float, int] = 0.5,
+        num_samples: int = 1,
+        **model_fit_kwargs,
+    ):
+        """Fit the underlying forecasting model (if applicable) and the fittable scorers, if any.
+
+        Train the model (if not already fitted and ``allow_model_training`` is set to True) and the
+        scorer(s) (if fittable) on the given time series.
+
+        Once the model is fitted, the series historical forecasts are computed,
+        representing what would have been forecasted by this model on the series.
+
+        The prediction and the series are then used to train the scorer(s).
+
+        Parameters
```

Comment:
the order of the parameters below does not follow the same order as in the function signature, could be a bit confusing when reviewing